### ISSUE TITLE
Feature: Async DataSource options for seeding

### ISSUE THREAD

**Issue by MartinDybdahlM** [tada5hi/typeorm-extension#707]
### Is your feature request related to a problem? Please describe.

You cannot use an async data source to seed the database. The seeder needs a sync data source, not a promise. If you give a file that exports a promise of a data source, the seeder will not work and will throw an error saying that it cannot get or load the database options.
```
OptionsError: The database options could not be located/loaded.
```

### Describe the solution you'd like

To be able to use a file that exports `Promise<DataSource>` for seeding, as this would allow DataSourceOptions to be fetched asynchronous. This would be more flexibility and scalability for seeding different environments.

### Describe alternatives you've considered

As a workaround we have now hardcoded the DataSourceOptions for seeding locally.

**Comment by javiermatias** [comment]
I have the same issue, but i have the options locally.. 
OptionsError: The database options could not be located/loaded.at Function.notFound .
I did the exact steps that are in the documentation.
this is my data-source.ts
OptionsError: The database options could not be located/loaded.
    at Function.notFound 

**this is my dataSource file**

import { DataSource, DataSourceOptions } from 'typeorm';
import { runSeeders, SeederOptions } from 'typeorm-extension';
import { User } from './registration/entities/user.entity';

(async () => {
  const options: DataSourceOptions & SeederOptions = {
    type: 'mysql',
    host: 'localhost',
    port: 3306,
    username: 'root',
    password: '123456',
    database: 'ausentismo',
    entities: [User],
    seeds: ['./*.seeder.ts'],
    factories: ['./*.factory.ts'],
  };

  const dataSource = new DataSource(options);
  await dataSource.initialize();

  await runSeeders(dataSource);
})();

**Comment by andrearruda82** [comment]
I have the same problem!

**Comment by tada5hi** [comment]
If I understood you correctly, you want to return an asynchronous function that resolves to an initialized data-source?

```typescript
export async function useDataSource() : DataSource {
    const options: DataSourceOptions & SeederOptions = {
    type: 'mysql',
    host: 'localhost',
    port: 3306,
    username: 'root',
    password: '123456',
    database: 'ausentismo',
    entities: [User],
    seeds: ['./.seeder.ts'],
    factories: ['./.factory.ts'],
    };
    
    const dataSource = new DataSource(options);
    await dataSource.initialize();

    return dataSource;
}
````

If you want to initialize the data source with a top level await, this would only be possible in an esm context.

**Comment by tada5hi** [comment]
@MartinDybdahlM @javiermatias @andrearruda82  ??

**Comment by MartinDybdahlM** [comment]
Sorry for the late reply... Thats exactly what i am trying to achieve. But when i try to seed using `typeorm-extension seed:run -d <path to datasource>` i get the error above

**Comment by tada5hi** [comment]
released in v3.1.0
